src/features/landing/components/AuthSection/index.tsx

```diff
@@ -23,9 +30,45 @@ function AuthSection() {
     formState: { errors },
   } = useForm<RegisterItem>();
   const onSubmit = (data: RegisterItem) => {
-
     console.log(data);
   };
+
+  const isLengthValid = (value: string) =>
+    value.length >= 8 && value.length <= 64;
+
+  const hasNumber = (value: string) => /[0-9]/.test(value);
+
+  const hasUppercase = (value: string) => /[A-Z]/.test(value);
+
+  function validatePassword(value: string) {
+    if (!isLengthValid(value))
```

Comment:
const isLengthValid = (value: string) =>
  value.length >= 8 && value.length <= 64;

const hasNumber = (value: string) => /[0-9]/.test(value);

const hasUppercase = (value: string) => /[A-Z]/.test(value);

const validatePassword = (value: string) => {
  const validations = [
    { condition: isLengthValid, message: 'Password must be between 8 and 64 characters long' },
    { condition: hasNumber, message: 'Password must contain at least one number' },
    { condition: hasUppercase, message: 'Password must contain at least one uppercase letter' },
  ];

  const invalidValidation = validations.find(validation => !validation.condition(value));

  return invalidValidation ? invalidValidation.message : true;
};

this code generate ChatGPT, your solution is to complicated, need refactor it and made you simpler